File: php/functions/format file size/formatfilesize.php
<?php

function format_file_size($bytes, $precision = 2) { 
    if ($bytes == 0) {
        return "n/a"; 
    }

    $units = array('B', 'KB', 'MB', 'GB', 'TB'); 
    $bytes = max($bytes, 0); 
    $pow = floor(($bytes ? log($bytes) : 0) / log(1024)); 
    $pow = min($pow, count($units) - 1); 
    $bytes /= (1 << (10 * $pow)); 

    return round($bytes, $precision) . '&nbsp;' . $units[$pow]; 
}

$file_size = 1234567890;
echo format_file_size($file_size);

File: rust/utils/format_file_size.rs
use std::env;

fn format_file_size(bytes: f64, precision: usize, nbsp: bool) -> String {
    if bytes == 0.0 { return "n/a".to_string(); }
    let units = ["B", "KB", "MB", "GB", "TB", "PB"];
    let mut pow = (bytes.ln() / 1024f64.ln()).floor() as usize;
    if pow > units.len() - 1 { pow = units.len() - 1; }
    let value = bytes / 1024f64.powi(pow as i32);
    let space = if nbsp { "\u{00A0}" } else { " " };
    format!("{:.*}{}{}", precision, value, space, units[pow])
}

fn main() {
    let args: Vec<String> = env::args().collect();
    if args.len() < 2 {
        eprintln!("usage: format_file_size <bytes> [precision]");
        std::process::exit(1);
    }
    let b: f64 = args[1].parse().unwrap_or(0.0);
    let p: usize = if args.len() > 2 { args[2].parse().unwrap_or(2) } else { 2 };
    println!("{}", format_file_size(b, p, true));
}


